graph/telemetry/istio/appender/service_entry.go

```diff
@@ -198,6 +200,10 @@ func (a ServiceEntryAppender) getServiceEntry(serviceName string, globalInfo *gr
 	}
 
 	for host, se := range serviceEntryHosts {
+		if !isExportedToNamespace(se, namespaceInfo) {
```

Comment:
related to my previous comment, I guess the `isExportedToNamespace` check should be moved above, before adding the SE to the hosts map

Reply:
@jotak Excellent discovery for the reported problem.  The issue with your suggested fix is that the code above is actually establishing a cached map (stored in globalnfo) that can be re-used for each namespace in the current graph.  So we can't exclude SEs in that code, it needs to have them all.  Let me take a look and see what can be done...

Reply:
@jotak pushed some changes that I think ensures complete maching when a host is duplicates and also guarantees a predictable/repeatable match.  Please give it a try when you get the chance.

Reply:
One of my scenarios doesn't seem to work:

-    with SE defined in same namespace and exportTo '.'

I expect to see the service entry in graph but it doesn't show up. Let me see if I find something in code...

Reply:
pff I think I've found the problem but it's going to be complex to solve. Perhaps we should merge this and create a follow-up because it actually fixes the initial issue, while uncovering a new one. Basically, the namespace that we need to consider for that service entry node is the namespace of the workload that originated the request, if I'm correct. We can chat about that when you're available.

Reply:
it also means that, within the same graph, a single service host might be represented both as a service entry and not a service entry, if the requests it serves comes from different workloads of different namespaces and `exportTo` is `.`.

Reply:
OK, so I was writing some slightly complicated code that started dealing with the source nodes and such when I realized that in fact the source workload namespaces really should, I think, always be the same as the namespace being processed by the appender.  So, with that simplification I submitted a commit for your consideration.  If it's not a valid assumption then I'll have to go back and reconsider.

Reply:
cool, fingers crossed, i'll test it